hydra_python_core/doc_writer.py

```diff
@@ -629,6 +632,46 @@ def generate(self) -> Dict[str, Any]:
         return error
 
 
+class HydraLink():
```

Comment:
Hydra:Link is a subclass of Hydra:Class, we can inherit it here and use the functions from Hydra:Class.

Reply:
If I inherit HydraClass then HydraLink will also have supportedProperty and functions related to it, but in the Spec `supportedProperty` does not explicitly define `HydraLink` as its domain.( But `supportedOperation` has `hydra:Link` as domain hence I have added it in the HydraLink class.). Having `supportedProperty` and related operations on a link doesn't make sense to spec-wise so I created a new class.

Reply:
We can inherit the class and only use the attributes/functions that we need. It is clearly defined in the spec that hydra:Link is a subclass of hydra:Resource. So spec wise I think it makes sense to inherit the class. Also supportedProperty does define hydra:Resource in its domain, since hydra:Link is a sub class of Resource, it is implied that it will also be in the domain.

Reply:
>We can inherit the class and only use the attributes/functions that we need.
Even if we do that it would be really misleading, as the spec doesn't mention anything about supportedProperties of Links.
>Also supportedProperty does define hydra:Resource in its domain,
Please can you point me to the statement, I couldn't locate it, what I can see is `supportedProperty` only defines `hydra:Class` as its domain. 

If we go subclassing every element which has `@type`=`hydra:Class` then HydraClassProp and HydraClassOperation which basically creates `hydra:SupportedProperty` and `hydra:SupportedOperation` respecively, will also need to be subclassed, do you think it'll make sense  to have `supportedProperties` of a `SupportedProperty`?  I thought you guys had this in mind when you wrote doc_writer and avoided such 'inheritance'.